pytools/py_codegen.py

```diff
@@ -35,17 +35,24 @@
 
 
 class PythonCodeGenerator(CodeGeneratorBase):
-    def get_module(self, name=None, write_file=True):
+    def get_module(self, name=None):
         if name is None:
             name = "<generated code>"
 
         result_dict = {}
         source_text = self.get()
 
-        if write_file:
-            with open(name, "w") as f:
-                f.write(source_text)
-                f.write("\n")
+        # {{{ Handle Python's linecache
+
+        import linecache
+
+        if name in linecache.cache:
+            from warnings import warn
+            warn(f"Overwriting existing generated code in linecache: '{name}'.",
+                 stacklevel=2)
```

Comment:
Use `stacklevel=3` if this is called through `get_function`. (Maybe use an underscored parameter to tell the two apart?)

Reply:
Done in [68ed96a](https://github.com/inducer/pytools/pull/295/commits/68ed96a881444eb48d7fc207a89c0ab5029cc7a7)